csrc/device_lower/pass/replace_size.cpp

```diff
@@ -230,20 +192,28 @@ void replaceSymbolicSizes(Fusion* fusion) {
 
   // Use a minimal number of sizes from provided tensors.
   auto extent_simplification_map = getSimplificationMap(fusion);
+
+  // We now need to map replacement scalars to their targets in tensor_dim_map
+  // if they exist. To do this we compose extent_simplification_map with
+  // tensor_dim_map.
   for (auto extent_entry : extent_simplification_map) {
     auto orig_extent = extent_entry.first;
     auto simplified_extent = extent_entry.second;
-    if (tensor_dim_map.count(orig_extent)) {
-      if (tensor_dim_map.count(simplified_extent)) {
-        tensor_dim_map[orig_extent] = tensor_dim_map[simplified_extent];
-      } else {
-        tensor_dim_map[orig_extent] = simplified_extent;
-      }
+    auto it = tensor_dim_map.find(simplified_extent);
+    if (it != tensor_dim_map.end()) {
+      extent_simplification_map[orig_extent] = it->second;
```

Comment:
nit: I always feel a little nervous when I see a map is modified while iterating it. I think this should be fine, though.